packages/dcos-net/extra/dcos-net-setup.py

```diff
@@ -15,110 +15,142 @@
         if result.stderr.strip().endswith(b'File exists'):
             result.returncode = 0
         return_code = result.returncode
     elif sys.argv[1] == 'iptables':
         # check whether a rule matching the specification does exist
         argv = ['-C' if arg in ['-A', '-I'] else arg for arg in sys.argv[1:]]
-        result = subprocess.run(argv)
+        result = run(argv)
         if result.returncode != 0:
             # if it doesn't exist append or insert that rules
-            result = subprocess.run(sys.argv[1:])
+            result = run(sys.argv[1:])
         return_code = result.returncode
     elif sys.argv[1] == '--ipv6':
         if os.getenv('DCOS_NET_IPV6', 'true') == 'false':
             sys.exit(0)
         else:
             del sys.argv[1]
-            result = subprocess.run(sys.argv)
+            result = run(sys.argv)
             return_code = result.returncode
     elif sys.argv[1:3] == ['networkd', 'add'] and len(sys.argv) == 4:
-        return_code = add_networkd_config_for_coreos(sys.argv[3])
+        return_code = add_networkd_config(sys.argv[3])
+    elif sys.argv[1:3] == ['networkmanager', 'add'] and len(sys.argv) == 4:
+        return_code = add_networkmanager_config(sys.argv[3])
     else:
-        result = subprocess.run(sys.argv[1:])
+        result = run(sys.argv[1:])
         return_code = result.returncode
     sys.exit(return_code)
 
 
-def add_networkd_config_for_coreos(src: str) -> int:
-    # systemd-networkd, when enabled, will wipe the configurations like IP
-    # address of network interfaces and this behavior happens only on coreos
-    # This problem is tracked by:
-    # https://jira.mesosphere.com/browse/DCOS_OSS-1790
-    # We need to mark interfaces managed by DC/OS as unmanaged when networkd is
-    # enabled on coreos
-    if platform.system() != "Linux" or "coreos" not in platform.release():
+def check_for_unit(unit: str) -> int:
+    result = run(['systemctl', 'list-unit-files', unit],
+                 stdout=subprocess.PIPE)
+
+    if result.returncode == 0 and unit in result.stdout.decode():
         return 0
 
-    networkd = b'systemd-networkd.service'
-    networkd_path = '/etc/systemd/network'
+    return result.returncode
+
+
+def is_unit_active(unit: str) -> bool:
+    result = run(['systemctl', 'is-active', unit],
+                 stdout=subprocess.PIPE)
+    if result.returncode == 0:
+        return True
+
+    return False
+
+
+def copy_file(src, dst) -> bool:
+    try:
+        if not filecmp.cmp(src, dst):
+            return bool(shutil.copyfile(src, dst))
+
+        return False
+    except FileNotFoundError:
+        return bool(shutil.copyfile(src, dst))
+
 
-    # Check if there is networkd
-    result = subprocess.run(['systemctl', 'list-unit-files', networkd],
-                            stdout=subprocess.PIPE)
+def add_config(unit: str, src: str, path: str) -> int:
+    # Check if the unit exists
+    result = check_for_unit(unit)
```

Comment:
`check_for_unit` returns an `int`, not a `subprocess` result. The following code can check `result != 0` directly.